redisinsight/api/src/modules/workbench/providers/workbench-commands.executor.ts

```diff
@@ -67,15 +73,20 @@ export class WorkbenchCommandsExecutor {
     dto: CreateCommandExecutionDto,
   ): Promise<CommandExecutionResult> {
     this.logger.log('Executing workbench command.');
-    const { command: commandLine } = dto;
+    const { command: commandLine, mode } = dto;
 
     try {
       const [command, ...args] = splitCliCommandLine(commandLine);
+
       const replyEncoding = checkHumanReadableCommands(`${command} ${args[0]}`) ? 'utf8' : undefined;
 
-      const response = this.formatter.format(
-        await this.redisTool.execCommand(clientOptions, command, args, replyEncoding),
-      );
+      const response = mode === WorkbenchMode.ASCII
+        ? this.rawFormatter.format(
+          await this.redisTool.execCommand(clientOptions, command, args, replyEncoding),
+        )
+        : this.utf8Formatter.format(
+          await this.redisTool.execCommand(clientOptions, command, args, replyEncoding),
+        );
```

Comment:
It will be better to have common interface for strategies and work with it like IFormatter.format. We should not check mode here.